src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/TickerTask.java

```diff
@@ -57,66 +74,169 @@
      *            The instance of our {@link SlimefunPlugin}
      */
     public void start(@Nonnull SlimefunPlugin plugin) {
+
         this.tickRate = SlimefunPlugin.getCfg().getInt("URID.custom-ticker-delay");
 
         BukkitScheduler scheduler = plugin.getServer().getScheduler();
-        scheduler.runTaskTimerAsynchronously(plugin, this, 100L, tickRate);
+        scheduler.runTaskLater(plugin, this, 100L);
     }
 
-    /**
-     * This method resets this {@link TickerTask} to run again.
-     */
-    public void reset() {
-        running = false;
+    @Override 
+    public final void run() {
+
+        // This should not happen. Only case where this may happen is if this task is scheduled or started multiple times.
+        if (running) {
+            return;
+        }
+        if (!Bukkit.isPrimaryThread()) {
+            return;
+        }
+
+        // Don't overload the Bukkit scheduler with a thousand runnables. It doesn't like that. This way, we can spread the
+        // time cost over several game ticks and adapt ticker rate to server performance.
+        final ArrayBlockingQueue<Runnable> syncTasks = new ArrayBlockingQueue<>(32);
+
+        // Iterate over the ticker tasks. Process the asynchronous ones on the Bukkit async task thread, and queue the sync
+        // tasks to the queue.
+        Bukkit.getScheduler().runTaskAsynchronously(
+                SlimefunPlugin.instance(),
+                () -> this.tick(syncTasks)
+        );
+
+        // Process the synchronous ticker tasks on the main thread.
+        processSyncTasks(syncTasks);
     }
 
-    @Override
-    public void run() {
+    private void processSyncTasks(final @Nonnull BlockingQueue<Runnable> tasks) {
+
+        if (!Bukkit.isPrimaryThread()) {
+            return;
+        }
+
+        // Let's not lag the main thread. Spikes suck. Steady time usage is much better.
+        long endNs = System.nanoTime() + 1_500_000L;
+
         try {
-            // If this method is actually still running... DON'T
-            if (running) {
-                return;
+            while (endNs > System.nanoTime()) {
+
+                // Don't wait on an empty queue for too long.
+                long s = System.nanoTime();
+                Runnable r = tasks.poll(250_000L, TimeUnit.NANOSECONDS);
```

Comment:
This magic number should definitely be a static final constant for easier tweaking.